sunpy/map/tests/test_mapbase.py

```diff
@@ -1653,3 +1654,43 @@ def test_plot_annotate_nonboolean(aia171_test_map):
     ax = plt.subplot(projection=aia171_test_map)
     with pytest.raises(TypeError, match="non-boolean value"):
         aia171_test_map.plot(ax)
+
+
+@pytest.fixture
+def aia171_test_dask_map(aia171_test_map):
+    dask_array = pytest.importorskip('dask.array')
+    return aia171_test_map._new_instance(
+        dask_array.from_array(aia171_test_map.data),
+        copy.deepcopy(aia171_test_map.meta)
+    )
+
+
+# This is needed for the reproject_to function
+with pytest.warns(VerifyWarning, match="Invalid 'BLANK' keyword in header."):
+    with fits.open(get_test_filepath('aia_171_level1.fits')) as hdu:
+        aia_wcs = astropy.wcs.WCS(header=hdu[0].header)
+
+
+@pytest.mark.parametrize(("func", "args"), [
+    ("max", {}),
+    ("mean", {}),
+    ("min", {}),
+    pytest.param("reproject_to", {"wcs": aia_wcs}, marks=pytest.mark.xfail(reason="reproject is not dask aware")),
+    pytest.param("resample", {"dimensions": (100, 100)*u.pix}, marks=pytest.mark.xfail()),
+    pytest.param("rotate", {}, marks=pytest.mark.xfail(reason="nanmedian is not implemented in Dask")),
+    ("std", {}),
+    ("superpixel", {"dimensions": (10, 10)*u.pix}),
+    ("submap", {"bottom_left": (100, 100)*u.pixel, "width": 10*u.pixel, "height":10*u.pixel}),
+])
+def test_method_preserves_dask_array(aia171_test_map, aia171_test_dask_map, func, args):
```

Comment:
```suggestion
def test_method_preserves_dask_array(aia171_test_map, aia171_test_dask_map, func, args):
    """
    Check that map methods preserve dask arrays if they are given as input, instead of eagerly
    operating on them and bringing them into memory.
    """
```